Frends.RabbitMQ.Read/Frends.RabbitMQ.Read.Tests/QuorumQueueTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Frends.RabbitMQ.Read.Definitions;
+using Frends.RabbitMQ.Read.Tests.Lib;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RabbitMQ.Client;
+using System.Text;
+
+namespace Frends.RabbitMQ.Read.Tests;
+
+[TestClass]
+public class QuorumQueueTests
+{
+    /// <summary>
+    /// You will need access to RabbitMQ queue, you can create it e.g. by running
+    /// docker run -d --hostname my-rabbit -p 5672:5672 -p 8080:1567 -p 15672:15672 -e RABBITMQ_DEFAULT_USER=agent -e RABBITMQ_DEFAULT_PASS=agent123  rabbitmq:3.9-management
+    /// In that case URI would be amqp://agent:agent123@localhost:5672 
+    /// Access UI from http://localhost:15672 username: agent, password: agent123
+    /// </summary>
+
+    private const string TestUri = "amqp://agent:agent123@localhost:5672";
+    private const string TestHost = "localhost";
+
+    [TestInitialize]
+    public void CreateExchangeAndQueue()
+    {
+        var factory = new ConnectionFactory { Uri = new Uri(TestUri) };
+        using var connection = factory.CreateConnection();
+        using var channel = connection.CreateModel();
+        channel.ExchangeDeclare("exchange", type: "fanout", durable: false, autoDelete: false);
+        var args = new Dictionary<string, object>();
+        args["x-queue-type"] = "quorum";
+        channel.QueueDeclare("quorumqueue", durable: true, exclusive: false, autoDelete: false, arguments: args);
+        channel.QueueBind("quorumqueue", "exchange", routingKey: "");
+    }
+
+    [TestCleanup]
+    public void DeleteExchangeAndQueue()
+    {
+        var factory = new ConnectionFactory { Uri = new Uri(TestUri) };
+        using var connection = factory.CreateConnection();
+        using var channel = connection.CreateModel();
+        channel.QueueDelete("quorumqueue", false, false);
+        channel.ExchangeDelete("exchange", ifUnused: false);
+    }
+
+    [TestMethod]
+    public void TestReadMultipleMessagesWithHostQuorum()
+    {
+        Connection connection = new()
+        {
+            Host = TestHost,
+            Username = "agent",
+            Password = "agent123",
+            RoutingKey = "quorumqueue",
+            QueueName = "quorumqueue",
+            AuthenticationMethod = AuthenticationMethod.Host,
+            ExchangeName = "exchange",
+
+            AutoAck = ReadAckType.AutoAck,
+            ReadMessageCount = 2,
+        };
+
+        Publish(connection, 2);
+        var result = RabbitMQ.Read(connection);
+
+        var test1 = result.MessagesBase64.Any(x => x.Headers.ContainsKey("X-AppId"));
+        var test2 = result.MessagesBase64.Any(x => x.Headers.ContainsValue("application id"));
+
+        Assert.IsTrue(result.MessagesBase64.Count == 2 && result.MessageUTF8.Count == 2 && result.Success);

```

Comment:
This check should be split into their own asserts and use `AreEqual` for equal comparisons.